routers/api/v1/repo_releases.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 The Gogs Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package v1
+
+import (
+	"errors"
+
+	api "github.com/gogits/go-gogs-client"
+
+	"github.com/gogits/gogs/modules/log"
+	"github.com/gogits/gogs/modules/middleware"
+	"github.com/gogits/gogs/models"
+)
+
+func ToApiRelease(release *models.Release, publisher *models.User) *api.Release {
+
+	return &api.Release{
+		Id:          		release.Id,
+		Publisher:			*ToApiUser(publisher),
+		TagName:			release.TagName,
+		LowerTagName:		release.LowerTagName,
+		Target:				release.Target,
+		Title:				release.Title,
+		Sha1:				release.Sha1,
+		NumCommits:			release.NumCommits,
+		Note:				release.Note,
+		IsDraft:			release.IsDraft,
+		IsPrerelease:		release.IsPrerelease,
+		Created:			release.Created,
+	}
+}
+
+func ListReleases(ctx *middleware.Context) {
+	rels, err := models.GetReleasesByRepoId(ctx.Repo.Repository.ID)
+	if err != nil {
+		log.Error(4, "GetReleasesByRepoId: %v", err)
+		ctx.Status(500)
+		return
+	}
+
+	apiReleases := make([]*api.Release, len(rels))
+	for i, rel := range rels {
+		publisher, err := models.GetUserByID(rel.PublisherId)
```

Comment:
It should check error see it's a db error or user not exist, if user not exist, make a fake user because release does exist, and needs to be returned.